npm/private/npm_translate_lock_state.bzl

```diff
@@ -328,53 +328,24 @@ def _write_action_cache(priv, rctx, label_store):
 def _copy_input_file(priv, rctx, label_store, key):
     if not label_store.has(key):
         fail("key not found '{}'".format(key))
-    if _is_text_file(label_store.path(key)):
-        _copy_text_file(priv, rctx, label_store, key)
-    else:
-        _copy_binary_file(priv, rctx, label_store, key)
-
-def _is_text_file(p):
-    # we can assume some files types are text files that are commonly copied
-    return p.endswith(".yaml") or p.endswith(".json") or p.endswith(".npmrc") or p.endswith(".lock")
 
-def _copy_text_file(priv, rctx, label_store, key):
+    # NB: rctx.read will convert binary files to text but that is acceptable for
+    # the purposes of calculating a hash of the file
     contents = rctx.read(label_store.path(key))
```

Comment:
Move this into the `if`?

Reply:
the read? it always needs to happen to calculate the hash

Reply:
ohh. I see what you mean.

Reply:
👍